### ISSUE TITLE
After receiving Too Many Requests error from kubernetes, the operator stops watching

### ISSUE THREAD

**Issue by wjsi** [nolar/kopf#962]
### Long story short

Kopf starts watching customresourcedefinitions and CRD objects at the same time, and it is quite frequent (5/10) that the latter watch is blocked when the server has limitations on request rates. Kopf stops `infinite_watch` call when this error is met, and operations over CRD objects are not handled.

### Kopf version

1.35.6

### Python version

3.9

### Logs

```none
/usr/local/lib/python3.9/site-packages/kopf/_core/reactor/running.py:176: FutureWarning: Absence of either namespaces or cluster-wide flag will become an error soon. For now, switching to the cluster-wide mode for backward compatibility.
  warnings.warn("Absence of either namespaces or cluster-wide flag will become an error soon."
[2022-09-19 12:34:17,417] kopf._core.reactor.r [DEBUG   ] Starting Kopf 1.35.6.
[2022-09-19 12:34:17,418] kopf.activities.star [DEBUG   ] Activity 'startup_fn' is invoked.
[2022-09-19 12:34:17,418] kopf.activities.star [INFO    ] Activity 'startup_fn' succeeded.
[2022-09-19 12:34:17,418] kopf._core.engines.a [INFO    ] Initial authentication has been initiated.
[2022-09-19 12:34:17,419] kopf.activities.auth [DEBUG   ] Activity 'login_via_client' is invoked.
[2022-09-19 12:34:17,420] kopf.activities.auth [DEBUG   ] Client is configured in cluster with service account.
[2022-09-19 12:34:17,420] kopf.activities.auth [INFO    ] Activity 'login_via_client' succeeded.
[2022-09-19 12:34:17,420] kopf._core.engines.a [INFO    ] Initial authentication has finished.
[2022-09-19 12:34:17,509] kopf._cogs.clients.w [DEBUG   ] Starting the watch-stream for customresourcedefinitions.v1.apiextensions.k8s.io cluster-wide.
[2022-09-19 12:34:17,510] kopf._cogs.clients.w [DEBUG   ] Starting the watch-stream for instances.v1.super.com cluster-wide.
[2022-09-19 12:34:17,527] kopf._cogs.clients.w [DEBUG   ] Stopping the watch-stream for instances.v1.super.com cluster-wide.
[2022-09-19 12:34:17,527] kopf._core.reactor.o [ERROR   ] Watcher for instances.v1.super.com@none has failed: ('StorageError: etcd cache not ready, Code: 6, Key: /super.com/instances, ResourceVersion: 0, AdditionalErrorMsg: ', {'kind': 'Status', 'apiVersion': 'v1', 'metadata': {}, 'status': 'Failure', 'message': 'StorageError: etcd cache not ready, Code: 6, Key: /super.com/instances, ResourceVersion: 0, AdditionalErrorMsg: ', 'details': {'retryAfterSeconds': 1}, 'code': 429})
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/errors.py", line 148, in check_response
    response.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 1004, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 429, message='Too Many Requests', url=URL('https://22.10.0.1:443/apis/super.com/v1/instances')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/aiokits/aiotasks.py", line 108, in guard
    await coro
  File "/usr/local/lib/python3.9/site-packages/kopf/_core/reactor/queueing.py", line 175, in watcher
    async for raw_event in stream:
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/watching.py", line 82, in infinite_watch
    async for raw_event in stream:
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/watching.py", line 159, in continuous_watch
    objs, resource_version = await fetching.list_objs(
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/fetching.py", line 28, in list_objs
    rsp = await api.get(
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/api.py", line 111, in get
    response = await request(
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/auth.py", line 45, in wrapper
    return await fn(*args, **kwargs, context=context)
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/api.py", line 85, in request
    await errors.check_response(response)  # but do not parse it!
  File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/errors.py", line 150, in check_response
    raise cls(payload, status=response.status) from e
kopf._cogs.clients.errors.APIClientError: ('StorageError: etcd cache not ready, Code: 6, Key: /super.com/instances, ResourceVersion: 0, AdditionalErrorMsg: ', {'kind': 'Status', 'apiVersion': 'v1', 'metadata': {}, 'status': 'Failure', 'message': 'StorageError: etcd cache not ready, Code: 6, Key: /super.com/instances, ResourceVersion: 0, AdditionalErrorMsg: ', 'details': {'retryAfterSeconds': 1}, 'code': 429})
```

**Comment by codewest** [comment]
I've encountered this issue as well.  Thanks for the patch @wjsi -- I plan to test #963 shortly.

**Comment by psontag** [comment]
Hey @nolar,
We also started running into this issue. It looks like currently the only way to fix this is to restart the operator. 

Do you already have an ETA for when the next Kopf version will be released that contains the change in #963?
